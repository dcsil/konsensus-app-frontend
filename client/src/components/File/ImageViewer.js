import FilePreviewer from 'react-file-previewer';
import React, { useEffect, useState } from 'react';
import { Container, Col, Row } from 'reactstrap';
import OrgTable from 'components/Profile/OrgTable';
import AccessHistoryTable from './AccessHistoryTable';
import { getFileById } from 'components/Auth/authFunctions';

const ImageViewer = ({ fileUrl, fileId }) => {
  const [actions, setActions] = useState([]);

  useEffect(() => {
    getFileById(fileId, setActions);
  }, []);
  return (
    <div style={{ alignItems: 'left' }}>
      <Row>
        <Col>
          <FilePreviewer
            file={{
              url: fileUrl,
            }}
          />
        </Col>
        <Col>
          <AccessHistoryTable actions={[actions]} />
        </Col>
      </Row>
    </div>
  );
};

export default ImageViewer;
