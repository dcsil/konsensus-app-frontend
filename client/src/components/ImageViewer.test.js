import React from 'react';
import { shallow } from 'enzyme';
import ImageViewer from './File/ImageViewer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Test ImageViewer component', () => {
  const wrapper = shallow(<ImageViewer />);
  expect(wrapper).toMatchSnapshot();
});
