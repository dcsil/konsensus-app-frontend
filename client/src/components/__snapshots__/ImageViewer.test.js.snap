// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ImageViewer component 1`] = `ShallowWrapper {}`;
